### PR TITLE
[IMP] 12.0 - Show UserError and ValidationError generating excel report

### PR DESCRIPTION
**Pull request by OpenCode**
https://github.com/OCA/reporting-engine/issues/350

### CODE REVIEW

File: report_xlsx/controllers/main.py [hunk 1]
```diff
@@ -3,39 +3,45 @@
...
+                    docids, data=data
+                )[0]
+                report_name = report.report_file
+                if report.print_report_name and not len(docids) > 1:
+                    obj = request.env[report.model].browse(docids[0])
+                    report_name = safe_eval(report.print_report_name,
+                                            {'object': obj, 'time': time})
+            except (UserError, ValidationError) as odoo_error:
+                raise werkzeug.exceptions.HTTPException(
+                    description=f'{odoo_error.name}. {odoo_error.value}')
```
Comment (gdgellatly) [inline report_xlsx/controllers/main.py]: Please don't use f strings for modules whose minimum supported version doesn't support them.

Comment (Saran440) [inline report_xlsx/controllers/main.py]: Functional Test.
Case `Exception` is not showed error.
you should coding `except Exception as e :` it's cover all case.

Comment (OpenCode) [reply report_xlsx/controllers/main.py]: ```except Exception as e``` is too generic and a bad pratic. I need to intercept only user errors so if we use UserError or ValidationError in our code, the messages can be showed. All the others errors (implementation errors) can be found in odoo server log.

Comment (OpenCode) [reply report_xlsx/controllers/main.py]: Good idea! Done!

Comment (Saran440) [reply report_xlsx/controllers/main.py]: okay, I see.

**Comment by Saran440** [comment]
@OpenCode Please check runbot and Travis status.

**Comment by OpenCode** [comment]
> @OpenCode Please check runbot and Travis status.

Travis is broken because of pip

ImportError: cannot import name 'SourceDistribution' from 'pip._internal.distributions.source'

@pedrobaeza Do you know I can I solve it?

**Comment by pedrobaeza** [comment]
Force push for triggering another build to see. Seems a temporary Travis glitch.

**Comment by OpenCode** [comment]
> @OpenCode Please check runbot and Travis status.

@Saran440 Travis and Runbot are green, now :)

**Review by Saran440 — APPROVED** [review]
Functional Test. LGTM :+1:

**Review by gdgellatly — APPROVED** [review]

**Comment by OpenCode** [comment]
@OCA/reporting-engine-maintainers Is it possible to merge this PR?

**Review by etobella — APPROVED** [review]
LGTM :+1:

**Pull request merged**